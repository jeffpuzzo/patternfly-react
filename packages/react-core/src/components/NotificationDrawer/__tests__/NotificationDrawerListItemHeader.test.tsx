import React from 'react';

import { render, screen } from '@testing-library/react';

import BellIcon from '@patternfly/react-icons/dist/esm/icons/bell-icon';
import { NotificationDrawerListItemHeader } from '../NotificationDrawerListItemHeader';

describe('NotificationDrawerListItemHeader', () => {
  test('renders with PatternFly Core styles', () => {
    render(<NotificationDrawerListItemHeader title="Pod quit unexpectedly" />);
    expect(screen.getByText('Pod quit unexpectedly').outerHTML).toMatchSnapshot();
  });

  test('className is added to the root element', () => {
    render(<NotificationDrawerListItemHeader title="Pod quit unexpectedly" className="extra-class" />);
    expect(screen.getByText('Pod quit unexpectedly').className).toMatchSnapshot();
  });

  test('list item header with custom icon applied ', () => {
    render(<NotificationDrawerListItemHeader title="Pod quit unexpectedly" icon={<BellIcon />} />);
    expect(screen.getByText('Pod quit unexpectedly').className).toMatchSnapshot();
  });

  test('list item header with srTitle applied ', () => {
    render(<NotificationDrawerListItemHeader title="Pod quit unexpectedly" srTitle="screen reader title" />);
    expect(screen.getByText('Pod quit unexpectedly').className).toMatchSnapshot();
  });

  test('list item header with variant applied ', () => {
    render(<NotificationDrawerListItemHeader title="Pod quit unexpectedly" variant="success" />);
    expect(screen.getByText('Pod quit unexpectedly').className).toMatchSnapshot();
  });

  test('list item header with truncateTitle', () => {
    render(<NotificationDrawerListItemHeader truncateTitle={1} title="Pod quit unexpectedly" variant="success" />);

    expect(screen.getByText('Pod quit unexpectedly').className).toContain('pf-m-truncate');
  });
});
