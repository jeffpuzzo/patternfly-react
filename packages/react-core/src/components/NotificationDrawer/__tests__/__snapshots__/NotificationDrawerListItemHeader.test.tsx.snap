// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationDrawerListItemHeader className is added to the root element 1`] = `"pf-c-notification-drawer__list-item-header-title"`;

exports[`NotificationDrawerListItemHeader list item header with custom icon applied  1`] = `"pf-c-notification-drawer__list-item-header-title"`;

exports[`NotificationDrawerListItemHeader list item header with srTitle applied  1`] = `"pf-c-notification-drawer__list-item-header-title"`;

exports[`NotificationDrawerListItemHeader list item header with variant applied  1`] = `"pf-c-notification-drawer__list-item-header-title"`;

exports[`NotificationDrawerListItemHeader renders with PatternFly Core styles 1`] = `"<h2 class=\\"pf-c-notification-drawer__list-item-header-title\\">Pod quit unexpectedly</h2>"`;
